### DISCUSSION TITLE
Targets: Mojo in the WEB

### DISCUSSION THREAD

**Discussion by ego** [modular/modular#78] [Ideas]
A huge audience is the Internet space and web development.
We need a tool to do great job not only on the backend side, but also on the frontend too.

MojoScript as a compiler that targets frontend: JavaScript/TypeScript like ClojureScript, Dart with Flatter, Kotlin with Multiplatform, Zig Bun, Vlang, Nim, Haxe, Wasm, ...

The future:
Mojo = Python + NodeJs =)

**Comment by TZDak** [comment]
Interesting proposal. It should be interesting to see what other people add to that idea and how the conversation evolves.

**Comment by sa-** [comment]
LLVM supports wasm as a target - you can see rust as an example. It would indeed be nice to see mojo used instead of js in the frontend :)

**Reply by ego** [reply]
> LLVM supports wasm as a target - you can see rust as an example. It would indeed be nice to see mojo used instead of js in the frontend :)

Yes, this is a fact. But there are many problems with wasm and it is not currently an industry standard for the frontend.
Bun is trying to bit up nodejs, but zig only release: 0.10.1, so it's very hard to sell it to businesses as a solution for the future.

One language for the backend with many features to optimize AI calculations ("strong" devs) and one language for the web for more "light" devs aka JavaScript. This can easily open a window for cloud engineering aka AWS CDK (via https://github.com/aws/jsii)

**Comment by timmermansjoy** [comment]
I think this will be something for the community to build out once the language is available. Same like node.js is developed by javascript and Django not by python

**Comment by lattner** [comment]
Lets track this as part of #71
